c_glib/arrow-glib/memory-pool.cpp

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <arrow-glib/memory-pool.hpp>
+
+G_BEGIN_DECLS
+
+/**
+ * SECTION: memory-pool
+ * @section_id: memory-pool-classes
+ * @title: memory-pool classes
+ * @include: arrow-glib/arrow-glib.h
+ *
+ * #GArrowMemoryPool is a class for memory allocation.
+ */
+
+typedef struct GArrowMemoryPoolPrivate_ {
+  arrow::MemoryPool* memory_pool;
```

Comment:
```suggestion
  arrow::MemoryPool *memory_pool;
```

We use C style in this file.